modules/git/git.go

```diff
@@ -164,46 +164,46 @@ func InitSimple(ctx context.Context) error {
 	return SetExecutablePath(setting.Git.Path)
 }
 
-var initOnce sync.Once
-
-// InitOnceWithSync initializes git module with version check and change global variables, sync gitconfig.
-// This method will update the global variables ONLY ONCE (just like git.CheckLFSVersion -- which is not ideal too),
-// otherwise there will be data-race problem at the moment.
-func InitOnceWithSync(ctx context.Context) (err error) {
+// InitWithSync initializes git module with version check and change global variables, sync gitconfig.
+// It should only be called at the beginning of the program initialization (TestMain/GlobalInitInstalled), otherwise there will be data-race problem.
+func InitWithSync(ctx context.Context) (err error) {
```

Comment:
The `WithSync` suffix here makes me think that this is synchronized and uses the sync. package.

It does not. I have a feeling it should just go back to being named `Init` or be named `FullInit`

Reply:
Done, now `git.Init`